File: app/src/main/java/com/blues/module_study/model_android/view/AndroidViewHolder.java
package com.blues.module_study.model_android.view;

import android.app.Activity;
import android.content.Intent;
import android.text.TextUtils;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.blues.R;
import com.blues.WebViewDialog;
import com.blues.module_study.ShowImageActivity;
import com.blues.module_study.model_android.model.AndroidEntity;
import com.bumptech.glide.Glide;
import com.jude.easyrecyclerview.adapter.BaseViewHolder;

import androidx.constraintlayout.widget.ConstraintLayout;
import androidx.core.app.ActivityOptionsCompat;

/**
 * User : Blues
 * Date : 2019/3/6
 */

public class AndroidViewHolder extends BaseViewHolder<AndroidEntity.ResultsBean> {

    private TextView tvDesc, tvAuthor, tvDate, tvPublisher;
    private ImageView ivCover;
    private ConstraintLayout clRoot;

    public AndroidViewHolder(ViewGroup parent) {
        super(parent, R.layout.item_android);

        tvDesc = $(R.id.tv_desc);
        tvAuthor = $(R.id.tv_author);
        clRoot = $(R.id.cl_root);
        ivCover = $(R.id.iv_cover);
        tvDate = $(R.id.tv_publish_time);
        tvPublisher = $(R.id.tv_source);
    }

    @Override
    public void setData(AndroidEntity.ResultsBean data) {
        super.setData(data);

        tvDesc.setText(data.getDesc());
        tvAuthor.setText(data.getWho());

        if (data.getImages() != null && data.getImages().size() != 0) {
            ivCover.setVisibility(View.VISIBLE);
            Glide.with(getContext()).load(data.getImages().get(0)).placeholder(R.mipmap.ic_img_error).into(ivCover);

            ivCover.setOnClickListener(v -> {
                //跳转到ShowImageActivity
                Intent intent = new Intent(getContext(), ShowImageActivity.class);
                intent.putExtra("param", data);
                ActivityOptionsCompat options = ActivityOptionsCompat.makeSceneTransitionAnimation((Activity) getContext(), ivCover, "transitionImage");
                getContext().startActivity(intent, options.toBundle());
            });
        }

        tvDate.setText(String.format("日期：%s", data.getPublishedAt().substring(0, data.getPublishedAt().indexOf("T"))));
        tvPublisher.setText(String.format("来自：%s", data.getSource()));

        if (!TextUtils.isEmpty(data.getUrl())) {
            clRoot.setOnClickListener(v -> new WebViewDialog(getContext(), data.getUrl()).show());
        }
    }
}

File: app/src/main/java/com/blues/gankio/v1/view/AndroidViewHolder.kt
package com.blues.gankio.v1.view

import android.view.ViewGroup
import com.blues.R
import android.widget.TextView
import androidx.constraintlayout.widget.ConstraintLayout
import android.content.Intent
import androidx.core.app.ActivityOptionsCompat
import android.app.Activity
import android.text.TextUtils
import android.view.View
import android.widget.ImageView
import coil.load
import com.blues.framework.dialog.WebViewDialog
import com.blues.gankio.v1.model.GankBean
import com.jude.easyrecyclerview.adapter.BaseViewHolder

/**
 * User : Blues
 * Date : 2019/3/6
 */
class AndroidViewHolder(parent: ViewGroup?) :
    BaseViewHolder<GankBean.ResultsBean>(parent, R.layout.item_android) {

    private val tvDesc: TextView = itemView.findViewById(R.id.tv_desc)
    private val tvAuthor: TextView = itemView.findViewById(R.id.tv_author)
    private val tvDate: TextView = itemView.findViewById(R.id.tv_publish_time)
    private val tvPublisher: TextView = itemView.findViewById(R.id.tv_source)
    private val ivCover: ImageView = itemView.findViewById(R.id.iv_cover)
    private val clRoot: ConstraintLayout = itemView.findViewById(R.id.cl_root)

    override fun setData(data: GankBean.ResultsBean) {
        super.setData(data)
        tvDesc.text = data.desc
        tvAuthor.text = data.who
        if (data.images.isNotEmpty()) {
            ivCover.visibility = View.VISIBLE
            ivCover.load(data.images[0]) {
                placeholder(R.mipmap.ic_img_error)
                error(R.mipmap.ic_img_error)
            }

            ivCover.setOnClickListener {   //跳转到ShowImageActivity
                val intent = Intent(context, ShowImageActivity::class.java).putExtra("param", data)
                val options = ActivityOptionsCompat.makeSceneTransitionAnimation((context as Activity), ivCover, "transitionImage")
                context.startActivity(intent, options.toBundle())
            }
        }
        tvDate.text = "日期：${data.publishedAt.substring(0, data.publishedAt.indexOf("T"))}"
        tvPublisher.text = "来自：${data.source}"

        if (!TextUtils.isEmpty(data.url)) {
            clRoot.setOnClickListener {
                WebViewDialog(context, data.url).show()
            }
        }
    }
}